code/mathematical_algorithms/src/factorial/factorial.erl

```diff
@@ -1,5 +1,5 @@
 -module(factorial).
--export([fac/1]).
+-export([factorial/1]).
 
-fac(0) -> 1;
-fac(N) -> N * fac(N-1).
\ No newline at end of file
+factorial(0) -> 1;
```

Comment:
Is there any way to just return 1 for a negative initial argument?

Reply:
That would be 

```erlang
factorial(N) when N < 0 -> 1.
```

Reply:
Do you think it should be handled like so?
Should the answer be yes, I can refactor the code to include the advent of negative numbers.

Reply:
Yes, it should. You should probably handle all cases.

Reply:
Shouldn't `<0` cause a matcherror since factorial itself is undefined when `<0`?

Reply:
That works as well. I just want it to be handled.

Reply:
Well a matcherror is thrown/raised when there is no matching function head, so technically it is already handled.  :-)

Reply:
(This also applies to the **Elixir** implementation).

Currently:

```erlang
factorial(0) -> 1;
factorial(N) when N > 0 -> N * factorial(N - 1).
```

The way is defined right now, if you call `factorial(-1)`, there will be no matching function in this Erlang module, so an error will be thrown.

If we define:

```erlang
factorial(N) when N < 0 -> 1.
```

We'll be returning 1 for every negative number (which... is not the definition of factorial).

Reply:
It's up to you all, but at least for me, it makes more sense to keep the mathematical definition of factorial, not allowing the user to call factorial of negative numbers.

Reply:
Ok, what I thought would happen was that if you called the function with a negative number, then the program would just not work. (Infinite loop). However, it seems that Elixir handles this very well! (Sorry for my incompetence; I don't know Elixir very well, it's just that I am going by general programming paradigms that _all_ cases should be handled).

Reply:
Hey, @arnavb, no problem. Erlang and Elixir are not very common programming languages and they have some different concepts!

Reply:
Heh yep, they are fully immutable functional languages that also have exceptions, and running through a set of matchers with no match ends up raising a `MatchError`.  :-)

They are languages designed to get real work done, so they have a lot of ways of doing things and are in general robust.  :-)